python/tritonserver/_api/_response.py

```diff
@@ -395,22 +138,103 @@ def _from_tritonserver_inference_response(
                     name,
                     data_type,
                     shape,
-                    _data_ptr,
-                    _byte_size,
-                    _memory_type,
-                    _memory_type_id,
-                    memory_buffer,
+                    data_ptr,
+                    byte_size,
+                    memory_type,
+                    memory_type_id,
                 ) = response.output(output_index)
+                memory_buffer = MemoryBuffer(
+                    data_ptr=data_ptr,
+                    memory_type=memory_type,
+                    memory_type_id=memory_type_id,
+                    size=byte_size,
+                    owner=response,
+                )
                 tensor = Tensor(data_type, shape, memory_buffer)
-
                 outputs[name] = tensor
             result.outputs = outputs
         except Exception as e:
             error = InternalError(f"Unexpected error in creating response object: {e}")
             error.args += (result,)
             result.error = error
 
+        # TODO: [DLIS-7824] Allocate the requested output memory type directly in C++.
+        if output_memory_type is not None:
+            try:
+                outputs = {}
+                for name, tensor in result.outputs.items():
+                    outputs[name] = tensor.to_device(output_memory_type)
+                result.outputs = outputs
+            except Exception as e:
+                raise InvalidArgumentError(
+                    f"Memory type {output_memory_type} not supported: {e}"
+                )
+
         # TODO: support classification
         # values["classification_label"] = response.output_classification_label()
 
         return result
+
+
+class AsyncResponseIterator:
```

Comment:
If I understand correctly - does this remove the ability for someone to set a queue / async queue in addition to the iterators?

Reply:
yes